src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,581 @@
+    val storagePrefix = row.getAs(ConfigColumns.etl_storage_prefix.toString).toString
+    val workspaceId = row.getAs(ConfigColumns.workspace_id.toString).toString
+    val testDetails = s"""StorageAccessTest storage : ${storagePrefix}"""
+    try {
+      dbutils.fs.mkdirs(s"""${storagePrefix}/test_access""")
+      dbutils.fs.put(s"""${storagePrefix}/test_access/testwrite""", "This is a file in cloud storage.")
+      dbutils.fs.head(s"""${storagePrefix}/test_access/testwrite""")
+      dbutils.fs.rm(s"""${storagePrefix}/test_access""", true)
+      validationStatus.append(DeploymentValidationReport(true,
+        getSimpleMsg("Storage_Access"),
+        testDetails,
+        Some("SUCCESS"),
+        Some(workspaceId)
+      ))
+    }
+    catch {
+      case exception: Exception =>
+        val msg = s"""Unable to read/write/create file in the provided etl storage prefix"""
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        logger.log(Level.ERROR, fullMsg)
+        validationStatus.append(DeploymentValidationReport(false,
+          getSimpleMsg("Storage_Access"),
+          testDetails,
+          Some(fullMsg),
+          Some(workspaceId)
+        ))
+        if (fastFail) {
+          throw new BadConfigException(fullMsg)
+        }
+
+
+    }
+  }
+
+  /**
+   * Performs clusters/list api call to check the access to the workspace
+   *
+   * @param row
+   */
+  private[overwatch] def validateApiUrl(row: Row): Unit = {
+    val url = row.getAs(ConfigColumns.api_url.toString).toString
+    val scope = row.getAs(ConfigColumns.secret_scope.toString).toString
+    val patKey = row.getAs(ConfigColumns.secret_key_dbpat.toString).toString
+    val workspaceId = row.getAs(ConfigColumns.workspace_id.toString).toString
+    val testDetails = s"""WorkSpaceURLConnectivityTest APIURL:${url} DBPATWorkspaceScope:${scope} SecretKey_DBPAT:${patKey}"""
+    try {
+      val patToken = dbutils.secrets.get(scope = scope, key = patKey)
+      val apiEnv = ApiEnv(false, url, patToken, getClass.getPackage.getImplementationVersion)
+      val endPoint = "clusters/list"
+      ApiCallV2(apiEnv, endPoint).execute().asDF()
+      validationStatus.append(DeploymentValidationReport(true,
+        getSimpleMsg("APIURL_Connectivity"),
+        testDetails,
+        Some("SUCCESS"),
+        Some(workspaceId)
+      ))
+    } catch {
+      case exception: Exception =>
+        val msg = s"""No Data retrieved workspaceId:${workspaceId} APIURL:${url} DBPATWorkspaceScope:${scope} SecretKey_DBPAT:${patKey}"""
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        logger.log(Level.ERROR, fullMsg)
+        validationStatus.append(DeploymentValidationReport(false,
+          getSimpleMsg("APIURL_Connectivity"),
+          testDetails,
+          Some(fullMsg),
+          Some(workspaceId)
+        ))
+
+    }
+
+  }
+
+  private[overwatch] def validateEtlStorage(): Rule = {
```

Comment:
Looks like many of these rules could be lumped into better rule functions with input params such as 

validateDistinct(configCol: ConfigColumn)
validateNotNull(configCol: ConfigColumn) 
etc

Significantly reduces code